src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -111,7 +111,18 @@ private IMethodNode CreateMethodEntrypointNode(MethodDesc method, TypeDesc const
             MethodWithGCInfo localMethod = null;
             if (CompilationModuleGroup.ContainsMethodBody(method, false))
             {
-                localMethod = new MethodWithGCInfo(method, signatureContext);
+                if (method is EcmaMethod || method is MethodForInstantiatedType)
```

Comment:
This should be fixed in a way that `ContainsMethodBody` says false - otherwise there's a disconnect between what `CompilationGroup` thinks is part of the image and what we actually do. Tomas is fixing this up in his #6345.

Reply:
@trylek Do you think you're close to merging that PR? I will make a similar change here since I want to merge this before the weekend.

Reply:
I hope to be able to merge this in before EOD today. Michal has just approved the change, I'm retesting after resolving one last conflict and pushing for hopefully the last round of lab testing.

Reply:
I pasted in Tomas' proposed change for this, so hopefully it'll merge cleanly regardless of who wins the check-in race.